src/Sylius/Bundle/UiBundle/Form/Type/SecurityLoginType.php

```diff
@@ -21,28 +21,25 @@
 
 final class SecurityLoginType extends AbstractType
 {
-    /**
-     * {@inheritdoc}
-     */
     public function buildForm(FormBuilderInterface $builder, array $options): void
     {
         $builder
             ->add('_username', TextType::class, [
                 'label' => 'sylius.form.login.username',
+                'attr' => ['data-login-username' => ''],
```

Comment:
I think we should add it to the template instead.

`form_row(form._username, {'attr': {'data-login-username': ''}})`

In order to make it more extendable, let's add a new Twig function `sylius_form_test_attribute(attribute)` which would return `['attr' => [$attribute => '']]` in test environment and `[]` in any other environment.

This way you can define the additional attributes in an easy way:

`form_row(form._username, sylius_form_test_attribute('data-login-username'))`

Or in case that there's some configuration already defined:

`form_row(form._username, sylius_form_test_attribute('data-login-username')|merge({'other': 'config'}))`

Then the attributes could be removed from the PHP form types.

Reply:
You a right it is great idea 👍